### ISSUE TITLE
October Rain not compatible with PHP 7.2

### ISSUE THREAD

**Issue by xf039** [octobercms/october#4812]
- OctoberCMS Build: master & october-1.0.460
- PHP Version: 7.2

### Description:
```php artisan october:util set build```
```PHP Parse error:  syntax error, unexpected ':', expecting '{' in /octobercms/public/vendor/october/rain/src/Support/helpers.php on line 302```

```
   function http_build_url(array $url, array $replace = [], $flags = HTTP_URL_REPLACE, array &$newUrl = []): string
    {
        return \October\Rain\Router\UrlGenerator::buildUrl($url, $replace, $flags, $newUrl);
    }
```

### Steps To Reproduce:
php72 composer install

No composer installation warning.

**Comment by LukeTowers** [comment]
@xf039 why'd you close this?

**Comment by xf039** [comment]
@LukeTowers my default PHP version is `php54`. I ran composer as `php72`, some automated commands fall back to using `php` what resulted in `artisan october:util set build` being executed as php 5.4 instead of 7.2.

**Comment by Developer-Nijat** [comment]
Same error happened

October version: 471
Web Server PHP version: 7.3.6
Cpanel PHP version 7.4

Now i installed new app from softaculous , same error happens. What is problem? How to fix it ?

**Comment by LukeTowers** [comment]
Same error? What's the exact error message @Developer-Nijat

**Comment by Developer-Nijat** [comment]
> Same error? What's the exact error message @Developer-Nijat 

PHP Parse error: syntax error, unexpected ':', expecting '{' in /octobercms/public/vendor/october/rain/src/Support/helpers.php on line 301

I contacted with hosting administrator and they are updated Php version to 7.3 but problem not solved currently 😒

**Comment by LukeTowers** [comment]
@Developer-Nijat sounds like your webserver isn't properly using the correct PHP version, double check with your hosting administrator to verify that your server software is actually running PHP 7.2 at a minimum.

**Comment by Developer-Nijat** [comment]
> @Developer-Nijat sounds like your webserver isn't properly using the correct PHP version, double check with your hosting administrator to verify that your server software is actually running PHP 7.2 at a minimum.

Sure. Problem solved after upgrading php version to 7.3 for specific subdomain and main domain.

**Comment by josemojica023** [comment]
> > @Developer-Nijat sounds like your webserver isn't properly using the correct PHP version, double check with your hosting administrator to verify that your server software is actually running PHP 7.2 at a minimum.
> 
> Sure. Problem solved after upgrading php version to 7.3 for specific subdomain and main domain.

Thank you so much !
